### PR TITLE
ATS System Tests

### PR DESCRIPTION
**Pull request by mikeb01**
Adds small changes to make it possible to test ATS with the Java system tests.
- Pass through system properties to configure the driver correctly
- Ignore specific errors

### CODE REVIEW

File: aeron-system-tests/src/test/java/io/aeron/MinFlowControlSystemTest.java [hunk 1]
```diff
@@ -197,50 +213,79 @@ else if (Publication.NOT_CONNECTED == result)
...
-        publication = clientA.addPublication(MULTICAST_URI, STREAM_ID);
+        awaitConnected(subscriptionB);
 
-        while (!subscriptionA.isConnected() || !subscriptionB.isConnected() || !publication.isConnected())
-        {
-            Tests.yield();
-        }
+        awaitConnected(publication);
+
+//        while (!subscriptionA.isConnected() || !subscriptionB.isConnected() || !publication.isConnected())
```
Comment (mjpt777) [inline aeron-system-tests/src/test/java/io/aeron/MinFlowControlSystemTest.java:238]: Dead code.

File: aeron-system-tests/src/test/java/io/aeron/MinFlowControlSystemTest.java [hunk 2]
```diff
@@ -197,50 +213,79 @@ else if (Publication.NOT_CONNECTED == result)
...
             }
+
+            if (0 == workDone)
+            {
+                Tests.yieldingIdle(
+                    () -> "numMessagesToSend=" + numMessagesToSend + " numMessagesLeftToSend=" + numMessagesLeftToSend +
+                    " numFragmentsReadFromA=" + numFragmentsReadFromA + " numFragmentsReadFromB=" +
+                    numFragmentsReadFromB);
+            }
+//            else
```
Comment (tmontgomery) [inline aeron-system-tests/src/test/java/io/aeron/MinFlowControlSystemTest.java:282]: to be removed?

File: aeron-system-tests/src/test/java/io/aeron/MultiDestinationCastTest.java [hunk 3]
```diff
@@ -109,13 +108,16 @@ private void launch(final ErrorHandler errorHandler)
...
+        testWatcher.dataCollector().add(driverA.context().aeronDirectory());
+        testWatcher.dataCollector().add(driverB.context().aeronDirectory());
     }
 
     @AfterEach
     void closeEverything()
     {
         CloseHelper.closeAll(clientB, clientA, driverB, driverA);
-        IoUtil.delete(new File(ROOT_DIR), true);
+//        IoUtil.delete(new File(ROOT_DIR), true);
```
Comment (tmontgomery) [inline aeron-system-tests/src/test/java/io/aeron/MultiDestinationCastTest.java:120]: should be removed?

File: aeron-system-tests/src/test/java/io/aeron/TaggedFlowControlSystemTest.java [hunk 4]
```diff
@@ -249,12 +249,13 @@ void shouldRemoveDeadReceiver()
 
         launch();
 
-        subscriptionA = clientA.addSubscription(MULTICAST_URI + "|gtag=123", STREAM_ID);
-        subscriptionB = clientB.addSubscription(MULTICAST_URI + "|gtag=123", STREAM_ID);
         publication = clientA.addPublication(MULTICAST_URI + "|fc=tagged,g:123,t:1s", STREAM_ID);
 
+        subscriptionA = clientA.addSubscription(MULTICAST_URI + "|gtag=123", STREAM_ID);
```
Comment (tmontgomery) [inline aeron-system-tests/src/test/java/io/aeron/TaggedFlowControlSystemTest.java:254]: why the change in order?

Comment (mikeb01) [reply aeron-system-tests/src/test/java/io/aeron/TaggedFlowControlSystemTest.java:254]: This gives the most chance of both receivers being included in flow control.

Comment (mikeb01) [reply aeron-system-tests/src/test/java/io/aeron/TaggedFlowControlSystemTest.java:254]: I've been thinking about the potential of surfacing a num receivers counter for a publication so that we can see whether a receiver has been included in flow control before proceeding.
